PA.py

```diff
@@ -1,685 +1,700 @@
+import webbrowser
+import os
+import sys
+import smtplib
+from email.message import EmailMessage
+import pywhatkit                                   # pip install pywhatkit
+import MyAlarm                                     # user-defined
+import pyjokes                                     # pip install pyjokes
+from speedtest import Speedtest                    # pip install speedtest-cli
+from pywikihow import search_wikihow               # pip install pywikihow
+import pyautogui                                   # pip install pyAutoGUI
+import poetpy                                      # pip install poetpy
+import random
+from forex_python.converter import CurrencyRates   # pip install forex-python
+import requests                                    # pip install requests
+import bs4                                         # pip install beautifulsoup4
+import time
+import wolframalpha                                # pip install wolframalpha
+from quote import quote                            # pip install quote
+import winshell as winshell                        # pip install winshell
+from geopy.geocoders import Nominatim              # pip install geopy  and pip install geocoder
+from geopy import distance
+# --------------------
+
+engine = pyttsx3.init()
+
+def img_requests(txt):
+    response=requests.get("https://source.unsplash.com/random?{0}".format(txt))
+    file=open('sample_image.jpg','wb')
+    file.write(response.content)
+    img=Image.open(r"sample_image.jpg")
+    img.show()
+    file.close
+
+def fun_talk(audio):
+    engine.say(audio)
+    engine.runAndWait()
+
+
+def wish_user():
+   
+    hour = int(datetime.datetime.now().hour)
+    if hour >= 0 and hour < 12:
+        fun_talk("Good Morning !")
+
+    elif hour >= 12 and hour < 18:
+        fun_talk("Good Afternoon !")
+
+    else:
+        fun_talk("Good Evening !")
+
+    fun_talk("I am P.A. (Python Assistant). Tell me how may I help you.")
+
+
+def get_command():
+    
+
+    rec = sr.Recognizer()
+    with sr.Microphone() as source:
+        print("Listening...")
+        rec.pause_threshold = 1
+
+        audio = rec.listen(source)
+
+        try:
+            print("Recognizing...")
+            query = rec.recognize_google(audio, language='en-in')
+            print(f"User said: {query}\n")
+
+        except Exception as e:
+            # print(e)
+            print("Say that again please...")
+            return "None"
+        return query
+
+
+if '__name__'=='__main__':
+    
+    wish_user()
+
+    while True:
+        
+        query = get_command().lower()
+
+        if 'wikipedia' in query:
+            fun_talk('Searching Wikipedia')
+            query = query.replace("wikipedia", "")
+            results = wikipedia.summary(query, sentences=3)
+            fun_talk("According to Wikipedia")
+            print(results)
+            fun_talk(results)
+        elif 'images' in query:

```

Comment:
write the code for this functionality at the end of this file, and add that `img_requests()` function inside the condition or may be in a new file.